tests/artifact_utils.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package tests
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+
+	"github.com/mendersoftware/mender-artifact/artifact"
+	"github.com/mendersoftware/mender-artifact/awriter"
+	"github.com/mendersoftware/mender-artifact/handlers"
+)
+
+type ArtifactDepends artifact.ArtifactDepends
+type ArtifactProvides artifact.ArtifactProvides
+
+type artifactReadCloser struct {
+	*bytes.Reader
+}
+
+func (rc *artifactReadCloser) Close() error {
+	return nil
+}
+
+// Creates a test rootfs artifact (version 3) with rootfs content @data, and the
+// provided depends and provides.
+func CreateTestArtifactV3(artifactProvides *ArtifactProvides,
+	artifactDepends *ArtifactDepends,
+	typeProvides, typeDepends *map[string]interface{},
+	data string) (*artifactReadCloser, error) {
+	var artifactArgs *awriter.WriteArtifactArgs
+	buf := bytes.NewBuffer(nil)
+	compress := artifact.NewCompressorNone()
+	artifactWriter := awriter.NewWriter(buf, compress)
+	updateFile, err := createFakeUpdateFile(data)
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(updateFile)
+	u := handlers.NewRootfsV3(updateFile)
+
+	artifactArgs = &awriter.WriteArtifactArgs{
+		Format:   "mender",
+		Version:  3,
+		Provides: (*artifact.ArtifactProvides)(artifactProvides),
```

Comment:
Question: Isn't the typing of the artifact package type a little unnecessary if you have to type-cast it back like this?

Reply:
I made it so that is unnecessary to import the `mender-artifact/artifact` library to use this function. Since the local type is just an alias the casting is safe to perform.